### DISCUSSION TITLE
I have some old scripts.  Can we update them?

### DISCUSSION THREAD

**Discussion by nollinvoyd** [xiaoxiaoflood/firefox-scripts#182] [General Support]
I was using several scripts in the Movable.uc.js format a few years ago, perhaps not originally created here, until they stopped working, some time after Fx 71.  I didn't realize until recently these custom buttons types were currently available for use again.

I tested them.  One partially works:

ZoomScrollFirefoxButton_Movable.uc.js

this appears on the toolbar but doesn't work:

Snap BackButton_Movable.uc.js

this on does not appear on the toolbar:

PreferenceTrackerButton_Movable.uc.js
 
Would it be ok if I posted them here to see if they can be updated?

Also, I seem to remember in the past an "up/Down" or "Top/Bottom" instant scroll button.script, like the "Back To Top for Firefox" addon.  Am I wrong about this?

**Comment by xiaoxiaoflood** [comment]
> Would it be ok if I posted them here to see if they can be updated?

Yes, of course. Hopefully someone else will step up.

**Comment by nollinvoyd** [comment]
Here's Snap BackButton_Movable.uc.js.  Like I mentioned, it doesn't respond.

`(function () {
  if (location != "chrome://browser/content/browser.xhtml") return;

  /* CustomizableUI
     http://developer.mozilla.org/docs/Mozilla/JavaScript_code_modules/CustomizableUI.jsm
     http://developer.mozilla.org/docs/Mozilla/JavaScript_code_modules/CustomizableUI.jsm/API-provided_widgets
  */

  try {
    CustomizableUI.createWidget({
      id: "__unique_identifier_snap_back_button", // should match id below
      type: "custom",
      defaultArea: CustomizableUI.AREA_MENUBAR,
   // defaultArea: CustomizableUI.AREA_NAVBAR,
      onBuild: function (aDocument) {
        var XUL_NS = "http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul";
        var toolbaritem = aDocument.createElementNS(XUL_NS, "toolbarbutton");
        toolbaritem.onclick = function (aEvent) {
          var win = aEvent.target.ownerDocument.defaultView;
          if (aEvent.button == 0) {
            if (aEvent.shiftKey) {
            } else {
              // snap back
              getWebNavigation().gotoIndex(0);
            }
          } else if (aEvent.button == 1) {
            if (aEvent.shiftKey) {
            } else {
            }
          }
        };
        var dataUrl = "data:image/png;base64," +
          "iVBORw0KGgoAAAANSUhEUgAAADAAAAAwCAYAAABXAvmHAAAABmJLR0QA/wD/AP+gvaeTAA" +
          "AI00lEQVRoge2aS2hc1x3Gf/ehGUm2RzNK9UgzjjQjWWnaUtruCqHCbaEUGtkI0gpkQqEL" +
          "LdpF2i6ThaC7ENr10OKNBSkuXZQS6oQ6Mm5wMMVQjF1qzUQuSAlEz5Ej6zX3nq+L+5g7I1" +
          "kPx48uemDQ1eUb3d/3f5xz7r2C/48Hj7G3R754NP3Z/qNewz7qFw47xs6fOSlPvz+s/tyF" +
          "kUE88/ejXuexGBg7f+akWkzJwuo+jP7chZFBD6Yl5Y96rUduYOz8mZNKqWRhdRtzsP7chZ" +
          "FBz7KmJfLS0a/3SA3E8KLbGJD2d1CHV14SeggHj8zA2PkzJ2mC3w/o3IWRQd+yppHyMgH8" +
          "UzMwdv7MSdIqIbqhDv8goHMXRgZ9uzHykpB5CgYCeKuEgoY1BhAB/B5AAbwd17xU1z/xDI" +
          "y9HcGrGwljoL+7LwZp5j93MYKvl01fV3+gj8w8KQPjU6N521glKwFf6Ckw8OwgIoh+EujH" +
          "fzg7YDw7qPkw2oWeIoPPDpIsoydiYHxqNC/blIS6o4sWegoUe4tB2I3CUjIxvGOIa54Qvt" +
          "hTbKz/hzBgHfULP7n4Su+255WQnguaTvR3FShE8MClG+8iYxDCwk61Hk8PWVjpqGwK3QUK" +
          "PUWQEPDujUvEM5ERJmEq+dOY3efdo8CPT43mPc8vIfVIzfCBJrg4ITzp9hOtQxKpuGxieB" +
          "BR89JYRknI5vNNn0OX0PjUaN6yKRlMT3hF+ruisgEII0cwjVpY6fYT7UNAigR8MYYPIaKf" +
          "YQU1fvY2kvwcqoSS8NEfff4LfUEDGljzlzjhdIaRN2xsb6Rvf3xraKe2nWrQ9w4iQdVbIu" +
          "N2JgwEAWgwFGWHqLnr59+Zfic2d2AGxqdG85ZLyaAeKejLQneBwWdPIQOr/iIrtYUQHja3" +
          "N9K3P7k9tFPbSTXoe08hwaq3yGptIYbhAPhooQiXi/BcPUv79sD41GjejuGjMihS6CkgxK" +
          "q/SLW2CCaCX0/f/uT2UC2KfFTz3cVA74X6GD4JrTgIUcTDQtuVjWiRlPZp4gb4aPboqsOv" +
          "1Zaobi/G+rv3bqUXl9aGPM+vw3cVKXQH+mptierOYhRy5jb+HR/XZ09Fp3YdP5Mu7DLxQA" +
          "PjU6N5u4WSUQhvxEBXkb4o8rXFBniv5qUXlxPwRhS7ivR3J/QJ+ETtHAo+3ppEvyZmpV09" +
          "MD41nrdbrJKR1SMTiIv7wvvpxeV7Q55nUtHMUTgEvGL4aDO0D3z020EZePXiK89JWyVEjz" +
          "B1+O79In9vyG8qm/6ug+GTWdh93Ayf6BNEENhggogNvHrxlefALyGrx2DAF8WuwqHhMaLQ" +
          "VXgE8HuXVgzftOi5SXjJ6jUYFMIcCO/XI9+/C36hntrPCR99u2G9SBrY8R3btYwtiFe+YJ" +
          "ukYIo0fuIiJq7fZDSsSC+QkvojNuuu89HFdsNLwgG49adb9775o69e9nx/GJGRYHV9BRnI" +
          "Hc+Sdo6BYNusA2Bblt/W1lLd3NzO+r5cCVY+WwFB9niW1lh//3PC14/TbkfcB+XKTKMBgH" +
          "/+8V/rL/6weMWyrWFExhhYWV9BiOyxHGmnFYAtfwMQtm35ralUdXNrO+t7xpXEyvoKSGSP" +
          "ZUk7bQ36ekaODg+QdjJxGc1UykFliLoBgH/9ubz+4ssDV8AelkxGJoQCcsdypO02kNg2my" +
          "BwHMtva01VNza3s75vXBkFmUB76g+aafYzmHIz8S53prxHBpImvnxm4IrEsFAGI1burSIZ" +
          "csdzpJw2LGDbbADgOLbf3paq3r+/FZiQWLm3giSyoR5gJ9Q3GzkMPEDK7Qi/VzfAXgYiE1" +
          "85e+qKjPm2kZWRTJgJkT3RSYvTjg1BZFFoorW6vp4w8VlYfsc7SdltWBbs+Jsx/GH7IjhS" +
          "ooQOyEDSxKmX+6YtWcNCGRmxuraCEXRmsrRYrRgZav4WEvS2n/JzmWeqy2tLWc/zAhNrKx" +
          "hEZyYX6An1QFdrkXY3S7uTpc3N0eZkaXWztDodpN0O0m6WlNNB2s2QSsALmLkzE2+nH2gA" +
          "4M5f7t4fGumfxreDxhasfraKjKEz00mL3YYsUTNbtLtZWmzHz53orC6tLWc933clWA3LKZ" +
          "cJMhHoN2l3c0S70oZVNl4vgnLZa6c6M1OOp/F9DUQmXjjbPy2fYUFGxgSRlch15GixgkZt" +
          "sYNad1zX78zkqstryw2ZUKxvBRGYb4aPp/u9ttnR+iTu3KmX0KFv6kd+990eb5OSZPIY4S" +
          "OG8oP05fvjFdKEIJcuXwKLVPvxtiEs0tGN+qmTg/SH+jq0+Ov7l4if5jXf3DffyKvx5v7A" +
          "DCQz8fz3i9O2McOylMGHxeoSAJ0duRheQHm2jDHya9s7Vcd1s4CLEUvVZQTkOnINES7PVu" +
          "LNWX2FDzPR/AkfvxzYxHuNu5fu3u/7wcC0JW9YRhkklleXMYJcNke406AyWw4jhe/VdqqO" +
          "6wQmQr0UmQBU1ych977Rr2sOvCN70Lj82uVPv/Obb/3MximBepGo/KeCJAb6Boi24QDI8M" +
          "Hkja9977cvFVXTtKTnA30ZyYR6xXO6JD789Y0jPat6qCdz7//yw4/Vkv6pDPNRdCqzM1Tu" +
          "ztRnFmMwoZG//eKDWduxX5L4KNKXZ8uU7zZFvr44PF4DEGQCv3VC0jwYjGBmtsLMbCV4ot" +
          "j0qPC9X12d8zzntKSPCMujPFtJ1H+j/rEbALj8+uVPHcyEMZqLSqD8UQUUGGrGufrG1Tkj" +
          "77RQpa6PsvBQ/J///cDl169/2uY7E4i5oClN+IROe76wuPrGP+Zsj9OIyq4Z5klnIBrvTV" +
          "5b8HEn8M1c44PZvfVXJq/P72CdxmiXiaOOR/aO7NrktQWc2oRgLobf5yXf9cnr875TO41U" +
          "eervyKJxbfLmgrPlTWCYi0ppv3F98ua8an69nJ7GO7Lmce2tmwuubyaMrLkD3rICcP3Nm/" +
          "OWMaclq/JUmnivce2tmwtpowmkucPor795c95FpyVVHgfPQ49v/PyFI/2zx9df+1L/Y0L5" +
          "3x3/Bd6B+4nhNxQyAAAAAElFTkSuQmCC";
        var tooltip = [];
        var props = {
          id: "__unique_identifier_snap_back_button",
          class: "toolbarbutton-1 chromeclass-toolbar-additional",
          label: "Snap Back",
          tooltiptext: "Snap Back" + tooltip.join("\n"),
          style: 'list-style-image: url("' + dataUrl + '"); -moz-image-region: auto;',
        };
        for (var p in props) toolbaritem.setAttribute(p, props[p]);
        return toolbaritem;
      },
    });
  } catch (e) {
    Components.utils.reportError(e);
  };
})();`
